provisioning/device/src/client.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+import * as Provisioning from './interfaces';
+import { ClientStateMachine } from './client_state_machine';
+import * as dbg from 'debug';
+const debug = dbg('azure-device-provisioning:device-client');
+
+export class ProvisioningDeviceClient implements Provisioning.ClientConfiguration {
+
+  /**
+   * User-Agent string passed to the service as part of communication
+   */
+  userAgent: string = 'azure-iot-node-provisioning'; // TODO
```

Comment:
>string = 'azure-iot-node-provisioning'; // TODO [](start = 13, length = 47)

const packageJson = require('../package.json');

eg: for http: RestApiClient(config, packageJson.name + '/' + packageJson.version)